platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -297,6 +304,55 @@ @implementation NSExpression (MGLAdditions)
     return subexpressions;
 }
 
++ (instancetype)mgl_expressionForString:(NSString *)string {
+    return [NSExpression expressionForConstantValue:string];
+}
+
++ (instancetype)mgl_expressionForColor:(MGLColor *)color {
+    return [NSExpression expressionForConstantValue:color];
+}
++ (instancetype)mgl_expressionForValue:(NSValue *)value {
+    return [NSExpression expressionForConstantValue:value];
+}
+
++ (instancetype)mgl_expressionForTernaryFunction:(NSString *)conditionString trueExpression:(NSExpression *)trueExpression falseExpresssion:(NSExpression *)falseExpression {
+    NSString *funtionFormat = [NSString stringWithFormat:@"TERNARY(%@, %%@, %%@)", conditionString];
+    return [NSExpression expressionWithFormat:funtionFormat, trueExpression, falseExpression];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(MGLExpressionStyleFunction)function defaultValue:(NSValue *)value stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    return [NSExpression mgl_expressionForStepFunction:function
+                                     defaultExpression:[NSExpression mgl_expressionForValue:value]
+                                                 stops:stops];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(MGLExpressionStyleFunction)function defaultString:(NSString *)string stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    return [NSExpression mgl_expressionForStepFunction:function
+                                     defaultExpression:[NSExpression mgl_expressionForString:string]
+                                                 stops:stops];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(MGLExpressionStyleFunction)function defaultColor:(MGLColor *)color stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    return [NSExpression mgl_expressionForStepFunction:function
+                                     defaultExpression:[NSExpression mgl_expressionForColor:color]
+                                                 stops:stops];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(MGLExpressionStyleFunction)function defaultExpression:(NSExpression *)expression stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    NSString *functionFormat = [NSString stringWithFormat:@"FUNCTION(%@, 'mgl_stepWithMinimum:stops:', %%@, %%@, %%@)", function];
```

Comment:
The first argument to `FUNCTION()` is an expression representing the operand of the function. It may be a `$zoomLevel` variable expression or a key path expression or an entire interpolation expression. So the first parameter to this method needs to be an NSExpression.